src/codeCache.cpp

```diff
@@ -162,18 +162,20 @@ const char* CodeCache::binarySearch(const void* address, LinearAllocator* alloca
     if (low > 0 && (_blobs[low - 1]._start == _blobs[low - 1]._end || _blobs[low - 1]._end == address)) {
         return _blobs[low - 1]._name;
     }
-
+    char buf[512];
+    // add 1 for the null terminator
+    int n;
+    char *allocated;
     if (_add_build_ids && _build_id) {
-        char buf[512];
-        // add 1 for the null terminator
-        const int n = 1 + snprintf(buf, sizeof(buf), "%s %s+%p_[bid]", _name, _build_id, (void*)((char*)address - (char*)_min_address));
-        char *allocated;
-        // if snprintf didn't overflow, and we manage to allocate a suitable buffer (from the LinearAllocator, not malloc,
-        // since this is signal context and malloc can deadlock) - then use them.
-        if (n < sizeof(buf) && (allocated = (char*)allocator->alloc(n)) != NULL) {
-            memcpy(allocated, buf, n);
-            return allocated;
-        }
+        n = 1 + snprintf(buf, sizeof(buf), "(%s) %s+%p_[bid]", _name, _build_id, (void*)((char*)address - (char*)_min_address));
```

Comment:
```suggestion
        n = 1 + snprintf(buf, sizeof(buf), "(%s %s+%p)_[bid]", _name, _build_id, (void*)((char*)address - (char*)_min_address));
```

this is nicer I think